adagrams/game.py

```diff
@@ -1,11 +1,111 @@
+
+import random
+import string
+
 def draw_letters():
-    pass
+    letter_pool = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    hand = []
+    random_letter = ""
+    while len(hand) < 10:
+        random_letter = random.choice(string.ascii_uppercase)
+        if letter_pool[random_letter] > 0:
+            letter_pool[random_letter] = letter_pool[random_letter] - 1
+            hand.append(random_letter)
+        
+    return hand
```

Comment:
Overall this function looks good, but it doesn't fully address the weighted nature of the alphabet's randomness. Your random choice picks a letter randomly but each letter has the same weight so there is an equal chance of drawing every letter. Is there a way you could extend that dictionary into a list that holds every letter the correct number of times it appears to address this randomness?